### ISSUE TITLE
1px white edge appears when enable shadow

### ISSUE THREAD

**Issue by guopenghui** [tauri-apps/window-shadows#13]
![image](https://user-images.githubusercontent.com/13451866/156786371-2c2100a3-69f7-4b3b-bee1-1d4af343ddb1.png)

When these lines are commented, it works well.

![image](https://user-images.githubusercontent.com/13451866/156786083-60ade5f7-5f5f-44c7-9447-a1a328c614fd.png)

And enable shadow, and focus the window, 1px white edge appears on the right and bottom of window.

![image](https://user-images.githubusercontent.com/13451866/156787262-310d64c0-f03b-4a61-8528-ace8d5421f83.png)

![image](https://user-images.githubusercontent.com/13451866/156787601-663c9611-fb5c-4bcc-93ff-c0e942dd6ced.png)

+ This issue only happens when the backgound-color has opacity setting, pure color is fine.
+ Only happends when window is focused
+ I haven't  set any 1px border to it.

**Comment by amrbashir** [comment]
Are you on Windows? If so, there is no way to disable that white border unfortunately (Not that I know of). 
If you look at "Visual studio installer", which I am pretty sure they are using the same api we use, it also has that white border.

**Comment by SP2G50000** [comment]
>If so, there is no way to disable that white border unfortunately (Not that I know of).

From what I know there is a way to make the white border non-noticeable.

I inspected your implementation and I noticed when you called `DwmExtendFrameIntoClientArea` you applied the border to all sides of the window. From what I know, you only need to apply it to one side to make the native shadow appear.

In the example given in this issue, the border is only visible on the right and bottom sides of the window. If you applied the border to only the top (`cyTopHeight`) and not the other sides, you would not be able to see the border and you would still get shadows. This isn't a perfect solution, but it should be much better than the current one.

I'm not an expert on Windows APIs so I might be wrong, but still, worth a shot.

**Comment by amrbashir** [comment]
> In the example given in this issue, the border is only visible on the right and bottom sides of the window.

I think top and left borders are obscured by the `wry` webview that is laid in top of the win32 window.

Last time I checked, the 4 borders were visible when just using a plain win32 window through `winit` or `tao` without webviews.

> If you applied the border to only the top (cyTopHeight) and not the other sides, you would not be able to see the border and you would still get shadows.

Yeah I am aware of that but it will look inconsistent when using a plain win32 window through `winit` or `tao`. You will see one border at the top. That's why I made the decision to show it on all sides so it looks consistent.
